python/ray/_private/services.py

```diff
@@ -1488,7 +1499,7 @@ def start_dashboard(
             dashboard_url = ""
         return dashboard_url, process_info
     except Exception as e:
-        if require_dashboard:
+        if raise_on_failure:
             raise e from e
         else:
             logger.error(f"Failed to start the dashboard: {e}")
```

Comment:
dashboard -> api server? make sure other places are updated consistently

Reply:
Actually, I will just keep the dashboard in this PR (it is a little random from end users saying API server now when the full refactoring is not done. For example, the process name is dashboard.py). I will address this comment when we change all dashboard -> API server.